hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -301,10 +269,8 @@ def _define_setting(
         options=None,
         validator=None,
         show_default=True,
-        future_default=not_set,
         deprecation_message=None,
         deprecated_since=None,
-        hide_repr=not_set,
```

Comment:
I disagree here; `hide_repr` was only ever used for a single setting - `timeout`, because the default value still had an effect even while the setting was deprecated.

With our more recent policy of immediately making deprecated things into no-ops wherever possible, we're just never going to need it.

Reply:
Fair.